.github/workflows/pr-ci.yaml

```diff
@@ -0,0 +1,49 @@
+# This workflow will install dependencies, run tests for both plugin and library components
+name: CloudFormation CLI  Java Plugin Pull Request CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+      AWS_REGION: us-east-1
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8]
+        java: [8, 11]
+    runs-on: ubuntu-latest
```

Comment:
might as well test on [`macos-latest` and `windows-latest`](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources) here as well

Reply:
added. i think it might also be useful to test compatibility with published libraries and plugins i.e. since we are managing two different packages of software, we should test the library against the published  plugin and vice versa to ensure backwards compatibility. These things get missed since we test the built library in a given PR with the built plugin in the given PR

Reply:
seeing some macos failures trying to build the java library. going to leave out the multiple OS testing for now since this was just supposed to be a migration and create a followup so we can do this in a more targeted way where we aren't just running 18 jobs for one pr. ideally we'd only need to test compatibility of the plugin with the given OS